### DISCUSSION TITLE
Failed to open TCP connection RSpec + Capybara testing (Ruby on Rails)

### DISCUSSION THREAD

**Discussion by SaidRasinlic** [community/community#52722] [Programming Help] [answered]
Hola every1,

I've been struggling a lot recently to fix this issue. I really Googled a lot and couldn't find an answer, most of the related questions were left unanswered.

### Issue description:
I'm trying to simply test my cases with RSpec + Capybara with Selenium by running the command **_`rails test:system`_** (even _**`rspec spec`**_ outputs the same), but for some reason the Google tab which opens is blank, and I also get a TCP connection error printed out. I was trying to find where this issue came from but couldn't.

### Error images:

![Error output message](https://user-images.githubusercontent.com/57558480/231760468-29651200-0ff9-410e-8d87-bb94acb57927.png)

![Blank Chrome tab](https://user-images.githubusercontent.com/57558480/231760643-39b8fc30-919f-415b-9362-93dd9bf6ee6f.png)

![My Gemfile](https://user-images.githubusercontent.com/57558480/231760554-e415dffd-7786-48fa-96d9-005430ab5f93.png)

### Version details:

- OS: Windows 10
- Ruby version: 3.1.2
- Running a command **"rails webdrivers:chromedriver:version"** prints: **2023-04-13 12:11:12 INFO  Webdrivers chromedriver 112.0.5615.49**

### What I've tried so far:

**I tried to change the timeout from 60 seconds to above by overriding Capybara timeout variable, but it didn't work! I really don't know how to fix this, it's been a while since I've been stuck on this, and I can't find an answer anywhere. If someone knows what's going on here please let me know.**

Thanks in advance 🙌

**Comment by joejbh** [comment]
Did you figure out what to do?  I'm having a similar issue

**Reply by SaidRasinlic** [reply, marked as answer]
## Yes I did!

Solution: As you can see in the screenshot, line **61** should be **commented out** and please make sure you add a **break**  statement after it, the file location of **socket.rb** on my PC is **C:\Ruby31-x64\lib\ruby\3.1.0\socket.rb**. I think it's their bug that somehow passed unnoticed in their sockets, and I was too lazy to send them a pull request, so if you want, guys, please do it. I hope this fixes a problem for you as well as the others.

_Please comment below if this worked for you so I can close this and mark it as resolved!_

![Capybara-Solution](https://github.com/community/community/assets/57558480/81e9d874-b7c3-4261-a054-044cbc96ad07)

P.S: My eyes bled trying to solve this error back then 🤬
